### ISSUE TITLE
os-string as a typeclass?

### ISSUE THREAD

**Issue by domenkozar** [haskell/os-string#12]
I've started writing a typeclass for all conversions:

```haskell
class ToOS a b where
  toOS :: a -> b

instance ToOS PosixPath OsString where
  toOS = OsString 
  
...
```

Then I've realized that some of the conversions would raise an exception like `FilePath -> m PosixPath`.

On top of that, a lot of these conversions are `O(n)` and now each library has to implement all the different interfaces.

Doesn't that smell like there should be a `OsString` typeclass for common operations just like we have `Num`?

**Comment by hasufell** [comment]
Yes, I don't think this is generally a good idea. On top of that, the conversion `ByteString -> WindowsString` expects `[Word16]`. It is not a good interface. 

When we're dealing with `FilePath` we don't want to make implicit choices. The user has to decide what is the right encoding.
